source/extensions/filters/common/expr/context.cc

```diff
@@ -0,0 +1,88 @@
+#include "extensions/filters/common/expr/context.h"
+
+#include "absl/strings/numbers.h"
+#include "absl/time/time.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Filters {
+namespace Common {
+namespace Expr {
+
+namespace {
+
+absl::optional<CelValue> convertHeaderEntry(const Http::HeaderEntry* header) {
+  if (header == nullptr) {
+    return {};
+  }
+  return CelValue::CreateString(header->value().getStringView());
+}
+
+} // namespace
+
+absl::optional<CelValue> HeadersWrapper::operator[](CelValue key) const {
+  if (!key.IsString()) {
+    return {};
+  }
+  auto out = headers_.get(Http::LowerCaseString(std::string(key.StringOrDie().value())));
+  return convertHeaderEntry(out);
+}
+
+absl::optional<CelValue> RequestWrapper::operator[](CelValue key) const {
+  if (!key.IsString()) {
+    return {};
+  }
+  auto value = key.StringOrDie().value();
+  if (value == Path) {
+    return convertHeaderEntry(wrapper_.headers_.Path());
+  } else if (value == Host) {
+    return convertHeaderEntry(wrapper_.headers_.Host());
+  } else if (value == Scheme) {
+    return convertHeaderEntry(wrapper_.headers_.Scheme());
+  } else if (value == Method) {
+    return convertHeaderEntry(wrapper_.headers_.Method());
+  } else if (value == Referer) {
+    return convertHeaderEntry(wrapper_.headers_.Referer());
+  } else if (value == Headers) {
+    return CelValue::CreateMap(&wrapper_);
+  } else if (value == Time) {
+    return CelValue::CreateTimestamp(absl::FromChrono(info_.startTime()));
+  } else if (value == ID) {
+    return convertHeaderEntry(wrapper_.headers_.RequestId());
+  } else if (value == UserAgent) {
+    return convertHeaderEntry(wrapper_.headers_.UserAgent());
+  } else if (value == Size) {
```

Comment:
If this is comparing against the `ContentLength` header, why not rename it to `ContentLength` instead of `Size`?

Reply:
See the comment below. We'd want to use true request size if available. But it's not available in the request headers, and also not clear when it is available.